crates/oxc_parser/src/ts/statement.rs

```diff
@@ -497,12 +497,20 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_decorator(&mut self) -> Result<Decorator<'a>> {
-        self.bump_any(); // bump @
         let span = self.start_span();
+        self.bump_any(); // bump @
         let expr = self.with_context(Context::Decorator, Self::parse_lhs_expression)?;
         Ok(self.ast.decorator(self.end_span(span), expr))
     }
 
+    pub fn at_decorator_before_export(&mut self) -> bool {
+        self.lookahead(Self::at_decorator_before_export_worker)
+    }
+
+    fn at_decorator_before_export_worker(&mut self) -> bool {
+        if self.eat_decorators().is_err() { false } else { self.cur_kind() == Kind::Export }
```

Comment:
Do we want to inline this so it's clearer that this is within a rewind?

Reply:
Yes I think it's a good point. Let's go for it.